src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -557,6 +587,11 @@ private void AttributePrefix(IEnumerable<HtmlSymbol> whitespace, IEnumerable<Htm
                 // Output the attribute name, the equals and optional quote. Ex: foo="
                 Output(SpanKind.Markup);
 
+                if (quote == HtmlSymbolType.Unknown && whitespaceAfterEquals.Any())
```

Comment:
Got it. @dougbu you may be misreading the diff. This is the `else` of `if (attributeCanBeConditional)`